internal/middleware/consistency/consistency.go

```diff
@@ -74,7 +79,23 @@ func addRevisionToContextFromConsistency(ctx context.Context, req hasConsistency
 	var revision datastore.Revision
 	consistency := req.GetConsistency()
 
+	withOptionalCursor, hasOptionalCursor := req.(hasOptionalCursor)
+
 	switch {
+	case hasOptionalCursor && withOptionalCursor.GetOptionalCursor() != nil:
```

Comment:
doh I literally asked about this, but the initial change was all in a different commit - sorry 🙏🏻 LGTM! Maybe those 2 commits should be squashed together

I think it's important to add a big disclaimer to the V1 Proto API documentation that adding a cursor effectively ignores the tunable consistency argument in every RPC method. I don't think this would be surprising, but it changes quite a bit how consistency is specified - with one argument effectively overriding another entirely

Reply:
I plan to document it when we describe how to use cursors in terms of best practices